src/lib.rs

```diff
@@ -265,24 +282,20 @@ where
                     shortcut_text.push_str(&format!("`{}`", text));
                 }
                 if let Some(text_for_header) = state.text_for_header.as_mut() {
-                    let code = format!("{}{}{}", options.code_block_token, text, options.code_block_token);
+                    let code = format!("`{}`", text);
                     text_for_header.push_str(&code);
                 }
-                let (start, end) = if text.contains(options.code_block_token) {
-                    (
-                        String::from_iter([options.code_block_token, options.code_block_token, ' ']),
-                        String::from_iter([' ', options.code_block_token, options.code_block_token]),
-                    )
+                if text.chars().all(|ch| ch == ' ') {
```

Comment:
I have the impression that this `if` statement should be an `else if` instead? However, this matches the existing code so I didn't change it here.

Reply:
These are indeed tempting!

However, it appears that this `if` statement also has a return value, and the others have not, while having quite different side-effects as well. So it seems adequate to do that, but as always, maybe a test can be constructed that lures out incorrect behaviour that makes a change necessary.

Reply:
Yeah, the side-effects are different and that confused me :smile:  I poked a bit more and I think I see the logic now: `text_for_header` gets a copy of the serialized Markdown code — and it is later used to pretty print tables. Similar for `shortcut_text which is used in `finalize`.

However, having said this, it seems to me that those variables ought to have the correct number of backticks as well. At least the `shortcut_text`, since I think it will generate an incorrect set of links. Let me try to make a test for this behavior as well (if I'm right, I suggest fixing this in a followup PR).